src/DotNetWorker.Grpc/FunctionMetadata/DefaultFunctionMetadataProvider.cs

```diff
@@ -22,7 +24,12 @@ public DefaultFunctionMetadataProvider()
             deserializationOptions.PropertyNameCaseInsensitive = true;
         }
 
-        public virtual async Task<ImmutableArray<RpcFunctionMetadata>> GetFunctionMetadataAsync(string directory)
+        public virtual async Task<ImmutableArray<IFunctionMetadata>> GetFunctionMetadataAsync(string directory)
+        {
+            return await GetRpcFunctionMetadataAsync(directory);
+        }
+
+        public virtual async Task<ImmutableArray<IFunctionMetadata>> GetRpcFunctionMetadataAsync(string directory)
```

Comment:
Should this be public?

Reply:
The interface this class implements is public because the source-generator needs to be able to access that same interface. As a result, this method is enforced as public. In previous iterations though, this was an internal method because source-gen hadn't been implemented yet.